src/Datadog.Trace/HttpOverStreams/DatadogHttpClient.cs

```diff
@@ -0,0 +1,99 @@
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.HttpOverStreams.HttpContent;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.HttpOverStreams
+{
+    internal class DatadogHttpClient
+    {
+        private const int BufferSize = 10240;
+        private static readonly Vendors.Serilog.ILogger Logger = DatadogLogging.For<DatadogHttpClient>();
+
+        public Task<HttpResponse> SendAsync(HttpRequest request, Stream requestStream, Stream responseStream)
+        {
+            Task.Run(() => SendRequest(request, requestStream));
```

Comment:
Why the `Task.Run`?

Reply:
Theoretically the server can reply as soon as the first byte is received, but this may not be necessary as it is against an HTTP Server.
I sourced this idea from Docker's HttpOverStreams implementation.
I will look into this and see if it's necessary or useful to keep around.

Reply:
The agent cannot start sending the response contents until we have sent the entire request contents. Even if the agent started sending header bytes right away (it doesn't), those bytes are buffered by the network stack and we can read them after we're done sending the request.

Writing and reading in parallel like this would be useful if we were streaming data to the server and the server was streaming data back to us at the same time, but we're not doing that.

Also, this fire-and-forget code is not handling exceptions that occur in `SendRequest()`.

Reply:
Removed the Task.Run, marking resolved.